### PR TITLE
Create the grafana policy at post-install

### LINKED ISSUE #6238: Grafana service fails to install on fresh cluster
When enabling grafana in `values.yaml` and trying to install it together with Istio, Helm throws the following error because the CRDs aren't there yet:

```sh
Error: apiVersion "authentication.istio.io/v1alpha1" in istio/charts/grafana/templates/service.yaml is not available
```

### PR DESCRIPTION
**Pull request by ymesika**
This is to avoid an error when installing Istio from scratch using Helm with the grafana enabled.

Fixes #6238

### CODE REVIEW

**Comment by ayj** [comment]
This should also fix https://github.com/istio/istio/issues/6803

File: install/kubernetes/helm/istio/charts/grafana/templates/create-custom-resources-job.yaml [hunk 1]
```diff
@@ -0,0 +1,79 @@
+{{- if .Values.global.mtls.enabled }}
```
Comment (costinm) [inline install/kubernetes/helm/istio/charts/grafana/templates/create-custom-resources-job.yaml]: Only for mtls ?

Comment (costinm) [reply install/kubernetes/helm/istio/charts/grafana/templates/create-custom-resources-job.yaml]: I mean - the policy must be defined even if global mtls is disabled - that is just the default but there are other ways to enable mtls ( per namespace, etc)

Comment (ymesika) [reply install/kubernetes/helm/istio/charts/grafana/templates/create-custom-resources-job.yaml]: @costinm Thanks. Will update it.

**Comment by christoph-puppe** [comment]
Hi

just had the ", unable to recognize "istio-demo-auth.yaml": no matches for config.istio.io/, Kind=metric, unable to recognize "istio-demo-auth.yaml": no matches for config.istio.io/, Kind=prometheus, unable to recognize "istio-demo-au"

with the current 1.0.2 following the tutorial at https://cloud.google.com/kubernetes-engine/docs/tutorials/installing-istio

**Pull request merged**